pure.zsh

```diff
@@ -210,21 +227,20 @@ prompt_pure_async_tasks() {
 	fi
 
 	# only perform tasks inside git working tree
-	[[ "${working_tree}" != "" ]] || return
+	[[ -n $working_tree ]] || return
 
-	if (( ${PURE_GIT_PULL:-1} )); then
-		# make sure working tree is not $HOME
-		[[ "${working_tree}" != "$HOME" ]] &&
+	# do not preform git fetch if it is disabled or working_tree == HOME
+	if (( ${PURE_GIT_PULL:-1} )) && [[ $working_tree != $HOME ]]; then
 		# tell worker to do a git fetch
-		async_job "prompt_pure" prompt_pure_async_git_fetch "$working_tree"
+		async_job "prompt_pure" prompt_pure_async_git_fetch "${working_tree}"
 	fi
 
 	# if dirty checking is sufficiently fast, tell worker to check it again, or wait for timeout
-	local time_since_last_dirty_check=$(( $EPOCHSECONDS - ${prompt_pure_git_last_dirty_check_timestamp:-0} ))
-	if (( $time_since_last_dirty_check > ${PURE_GIT_DELAY_DIRTY_CHECK:-1800} )); then
+	integer time_since_last_dirty_check=$(( EPOCHSECONDS - ${prompt_pure_git_last_dirty_check_timestamp:-0} ))
```

Comment:
Because all tokens in an arithmetic expression are recursively expanded (as variable names) for values. According to [official doc](http://zsh.sourceforge.net/Doc/Release/Arithmetic-Evaluation.html):

> Named parameters and subscripted arrays can be referenced by name within an arithmetic expression without using the parameter expansion syntax.

Canonical example:

```zsh
> foo=bar
> bar=5
> echo $((foo))
5
```

This is also the case in other shells, e.g., bash, by the way.